### PR TITLE
Cosine T_max Sweep: T_max={140, 180} vs baseline 160

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The cosine annealing schedule (`--cosine_T_max 160`) hasn't been tuned since the baseline architecture changed significantly (GSB, aft_foil_srf, PCGrad all added). T_max controls when LR reaches near-zero — currently it equals the epoch count exactly, so LR decays to ~0 precisely at convergence.

- **T_max=140** (shorter): LR decays faster, model commits to solution earlier. Potentially better generalization if the architecture finds a good basin quickly.
- **T_max=180** (longer): LR stays elevated longer, more gradient flow near convergence. Potentially better if the model benefits from sustained high-LR training with the current complexity.

This is a cheap, untested lever. Cosine T_max was never tuned after GSB/aft_foil_srf/PCGrad were added.

## Instructions

Run 4 seeds total (42 and 73 for each T_max variant). Use `--wandb_group` so runs are grouped.

**Variant A — T_max=140:**
```bash
cd cfd_tandemfoil && python train.py --agent thorfinn --seed 42 \
  --wandb_name "thorfinn/cosine-tmax-140-s42" --wandb_group "thorfinn/cosine-tmax-sweep" \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 140 --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3 \
  --aft_foil_srf --aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05 \
  --pcgrad_3way --pcgrad_extreme_pct 0.15 --gap_stagger_spatial_bias
```
Repeat with `--seed 73 --wandb_name "thorfinn/cosine-tmax-140-s73"`.

**Variant B — T_max=180:**
```bash
cd cfd_tandemfoil && python train.py --agent thorfinn --seed 42 \
  --wandb_name "thorfinn/cosine-tmax-180-s42" --wandb_group "thorfinn/cosine-tmax-sweep" \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 180 --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3 \
  --aft_foil_srf --aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05 \
  --pcgrad_3way --pcgrad_extreme_pct 0.15 --gap_stagger_spatial_bias
```
Repeat with `--seed 73 --wandb_name "thorfinn/cosine-tmax-180-s73"`.

Run all 4 in parallel. Only change is `--cosine_T_max`. Everything else identical to baseline.

## Baseline

- **p_in:** 13.05 | **p_oodc:** 7.70 | **p_tan:** 28.60 | **p_re:** 6.55
- W&B: d7l91p0x (seed 42), j9btfx09 (seed 73)

**Reproduce baseline:**
```bash
cd cfd_tandemfoil && python train.py --agent <name> --wandb_name "<name>/baseline-gsb-pcgrad" \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 160 --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3 \
  --aft_foil_srf --aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05 \
  --pcgrad_3way --pcgrad_extreme_pct 0.15 --gap_stagger_spatial_bias
```

## Report Format

| T_max | p_in avg | p_oodc avg | p_tan avg | p_re avg | W&B runs |
|-------|----------|------------|-----------|----------|----------|
| 140 | | | | | |
| 160 (baseline) | 13.05 | 7.70 | 28.60 | 6.55 | d7l91p0x, j9btfx09 |
| 180 | | | | | |

### CODE REVIEW

**Comment by tcapelle** [comment]
## Experiment Results: Cosine T_max Sweep

### Configuration
- T_max={140, 180} vs baseline T_max=160
- Seeds 42 and 73 for each variant
- All baseline flags: `--aft_foil_srf --aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05 --pcgrad_3way --pcgrad_extreme_pct 0.15 --gap_stagger_spatial_bias`
- 150-151 epochs reached (180-min timeout)
- W&B group: `thorfinn/cosine-tmax-sweep`

### Results

| T_max | Seed | p_in | p_tan | p_oodc | p_re | W&B |
|-------|------|------|-------|--------|------|-----|
| **160 (baseline)** | avg | **13.05** | **28.60** | **7.70** | **6.55** | d7l91p0x, j9btfx09 |
| 140 | 42 | 13.2 | 29.5 | 7.6 | 6.4 | [9cg2dt2a](https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/9cg2dt2a) |
| 140 | 73 | 12.7 | 29.3 | 7.9 | 6.5 | [scwsj91y](https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/scwsj91y) |
| **140 avg** | — | **12.95** | **29.40** | **7.75** | **6.45** | — |
| 180 | 42 | 13.5 | 29.6 | 7.6 | 6.7 | [3beaxpua](https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/3beaxpua) |
| 180 | 73 | 13.8 | 29.2 | 8.3 | 6.6 | [tz7aj1ji](https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/tz7aj1ji) |
| **180 avg** | — | **13.65** | **29.40** | **7.95** | **6.65** | — |

### Summary Comparison (2-seed avg)

| Metric | T=140 | T=160 (baseline) | T=180 |
|--------|-------|-------------------|-------|
| p_in   | **12.95** (-0.8%) | 13.05 | 13.65 (+4.6%) |
| p_tan  | 29.40 (+2.8%) | **28.60** | 29.40 (+2.8%) |
| p_oodc | 7.75 (+0.6%) | **7.70** | 7.95 (+3.2%) |
| p_re   | **6.45** (-1.5%) | 6.55 | 6.65 (+1.5%) |

### Analysis

**Negative result — T_max=160 (baseline) remains optimal for p_tan.**

Both T_max=140 and T_max=180 show identical p_tan regression (+2.8% vs baseline). The baseline T_max=160 appears to be well-tuned for the current architecture.

**T_max=140 observations:**
- Best p_in (12.95) and best p_re (6.45) — faster LR decay helps in-distribution and Re generalization slightly
- But p_tan is worse, suggesting the model needs sustained high LR to learn tandem transfer features

**T_max=180 observations:**
- Worst on all metrics except p_tan (tied with T=140)
- Longer LR warmth doesn't help — the model may be overfit by the time LR decays

**Conclusion:** T_max=160 is a good sweet spot. The LR schedule is not a bottleneck for current performance.

### Suggested Follow-ups
- No further T_max tuning recommended — the current value is well-matched to the training budget
- If more epochs become available (higher timeout), T_max should scale proportionally

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2154 because ADVISOR: Closing — T_max=160 confirmed optimal. Both alternatives regress p_tan identically.

**Results:** T_max=140 avg p_tan=29.40 (+2.8%), T_max=180 avg p_tan=29.40 (+2.8%). Only p_in and p_re showed marginal improvement with T_max=140 (12.95 vs 13.05, 6.45 vs 6.55), but not worth the p_tan regression.

**Key finding:** The LR schedule is well-tuned for the current architecture. The 160-epoch budget with T_max=160 (LR hits near-zero at the end) is the sweet spot. Neither faster decay (140) nor extended warmth (180) helps p_tan. Added to confirmed optimal hyperparameters.

Good systematic sweep, thorfinn. Moving you to a bolder architectural experiment..
